vividus-plugin-rest-api/src/main/java/org/vividus/archive/steps/ArchiveSteps.java

```diff
@@ -0,0 +1,127 @@
+        Map<String, byte[]> zipEntries = ZipUtils.readZipEntriesFromBytes(archiveData.getBytes(),
+                expectedEntries::contains);
+        parameters.forEach(arcVar ->
+        {
+            String path = arcVar.getPath();
+            Optional.ofNullable(zipEntries.get(path)).ifPresentOrElse(
+                    data -> variableContext.putVariable(arcVar.getScopes(), arcVar.getVariableName(),
+                            arcVar.getOutputFormat().convert(data)),
+                    () -> softAssert.recordFailedAssertion(
+                            String.format("Unable to find entry by name %s in response archive", path)));
+        });
+    }
+
+    /**
+     * Verifies that at least one (or no one) entry in an archive matches the specified string comparison rule.
+     * If comparison rule column does not exist,
+     * the verification that archive entries have the specified names is performed.
+     * <p>
+     * Usage example:
+     * </p>
+     * <p>
+     * <code>
+     * Then `${response-as-bytes}` archive contains entries with names:$parameters<br>
+     * |rule      |name                                    |<br>
+     * |contains  |2011-11-11/skyrim.json                  |<br>
+     * |matches   |files/2011-11-11/logs/papyrus\.\d+\.log |<br>
+     * </code>
+     * </p>
+     *
+     * @param archiveData The data of archive to verify
+     * @param parameters  The ExampleTable that contains specified string comparison <b>rule</b> and entry <b>name</b>
+     *                    pattern that should be found using current <b>rule</b>. Available columns:
+     *                    <ul>
+     *                    <li>rule - String comparison rule: "is equal to", "contains", "does not contain", "matches"
+     *                    .</li>
+     *                    <li>name - Desired entry name pattern used with current <b>rule</b>.</li>
+     *                    </ul>
+     */
+    @Then("`$archiveData` archive contains entries with names:$parameters")
+    public void verifyArchiveContainsEntries(DataWrapper archiveData, List<NamedEntry> parameters)
+    {
+        Set<String> entryNames = ZipUtils.readZipEntryNamesFromBytes(archiveData.getBytes());
+
+        parameters.forEach(entry ->
+        {
+            String expectedName = entry.getName();
+            if (entry.getRule() != null)
+            {
+                StringComparisonRule comparisonRule = entry.getRule();
+                softAssert.assertThat(String.format(
+                        "The response archive contains entry matching the comparison rule '%s' with name pattern '%s'",
+                        comparisonRule, expectedName), entryNames, hasItem(comparisonRule.createMatcher(expectedName)));
+            }
+            else
+            {
+                softAssert.assertThat("The response archive contains entry with name " + expectedName, entryNames,
```

Comment:
https://github.com/vividus-framework/vividus/pull/2797#discussion_r882372517 - the same here